CRM/Donrec/Exporters/EmailPDF.php

```diff
@@ -107,7 +107,7 @@ protected function postprocessPDF($file, $snapshot_receipt, $is_test) {
         civicrm_api3('Activity', 'create', array(
           'activity_type_id'   => $this->getEmailErrorActivityID(),
           'subject'            => E::ts("Donation receipt not delivered"),
-          'status_id'          => CRM_Core_OptionGroup::getValue('activity_status', 'Scheduled', 'name'),
+          'status_id'          => CRM_Core_OptionGroup::getValue('activity_status', 'Scheduled', 'name'), // TODO CRM_Core_PseudoConstant::getKey('BAO_NAME ???', 'activity_status', 'Scheduled')
```

Comment:
BAO name here is CRM_Activity_BAO_Activity and it would be status_id field

Reply:
thanks a lot, this is super helpful! :-) 

Is there a documentation where I could have found this?

Reply:
You could do a search of the core code to see how get Key is used e.g. https://github.com/civicrm/civicrm-core/blob/53a1e33da4c2fb904e3f6c8b68160ab76a36c30a/CRM/Case/Page/AJAX.php#L65

Reply:
We have now decided to go another way, and implemented our own drop-in replacement, see #125, changes here: https://github.com/systopia/de.systopia.donrec/compare/issue/125. 

This is currently being tested, but when it's done it should make this ticket here obsolete as well.